src/app/user/login/get.controller.js

```diff
@@ -1,7 +1,20 @@
 const navUtil = require('../../../common/utils/nav');
 
+/**
+ * If incoming object contains a dbId (id), vr (verified) and rl (role),
+ * then consider the user logged in.
+ *
+ * @param {Object} userSessionObject object representing the User in the session.
+ */
+const checkUserInSession = (userSessionObject) => {
+  if (userSessionObject) {
+    return userSessionObject.dbId && userSessionObject.vr && userSessionObject.rl;
+  }
+  return false;
+};
+
 module.exports = (req, res) => {
-  if (req.headers.referer && req.session.u && req.session.u.dbId && req.session.u.vr && req.session.u.rl) {
+  if (req.headers.referer && checkUserInSession(req.session.u)) {
```

Comment:
Sonarqube driven development, moving the boolean predicates into their own function.